src/usecases/FillDailyReport.spec.ts

```diff
@@ -32,21 +34,25 @@ describe("FillDailyReport", () => {
     new MemoryProducedSolarEnergyFetcher(producedSolarEnergy),
     new MemorySoldSolarEnergyFetcher(soldSolarEnergy),
     repository,
+    basePricesFinder,
   );
 
   it("should build and store the daily report", async () => {
     await sut.execute(new FillDailyReportCommand(day));
 
     assert.equal(repository.dailyReports.length, 1);
-    assert.deepEqual(
-      repository.dailyReports[0],
-      new DailyReport(
+    assert.deepEqual(repository.dailyReports[0], {
+      dailyReport: new DailyReport(
         day,
         previousYearConsumption,
         consumption,
         producedSolarEnergy,
         soldSolarEnergy,
       ),
-    );
+      basePrices: {
+        month: basePricesFinder.findForMonth(day.month),
+        day: basePricesFinder.findForDay(day),
+      },
```

Comment:
self note: this is weird, instead this method should probably receive a value object which would provide the same ⇨ BasePriceFinder would be simplified